packages/fiori-elements-writer/src/data/manifestSettings.ts

```diff
@@ -43,10 +43,16 @@ export function extendManifestJson<T>(
         const minVersion = semVer.coerce(feApp.ui5?.minUI5Version);
         if (!minVersion || semVer.gte(minVersion, '1.94.0')) {
             const entityConfig = (feApp.template.settings as LROPSettings).entityConfig as ManifestEntitySettings;
-            entityConfig.contextPath = `/${entityConfig.mainEntityName}`;
+            // Set the contextPath for the main entity
+            entityConfig.contextPath = entityConfig.mainEntityParameterName
+                ? `/${entityConfig.mainEntityName}/${entityConfig.mainEntityParameterName}`
+                : `/${entityConfig.mainEntityName}`;
 
+            // Set the contextPath for the navigation entity
             if (entityConfig.navigationEntity?.EntitySet) {
-                entityConfig.navigationEntity.contextPath = `${entityConfig.contextPath}/${entityConfig.navigationEntity.Name}`;
+                entityConfig.navigationEntity.contextPath = entityConfig.mainEntityParameterName
```

Comment:
I dont think theres any need to check `mainEntityParameterName` when creating the nav entity context path. It will always be :`/${entityConfig.mainEntityName}/${entityConfig.navigationEntity.Name}`

```suggestion
                entityConfig.navigationEntity.contextPath = `/${entityConfig.mainEntityName}/${entityConfig.navigationEntity.Name}`
```